### DISCUSSION TITLE
CSS variables are not valid values for `StrictableProps` when `strictPropertyValues` is enabled

### DISCUSSION THREAD

**Discussion by etlovett** [chakra-ui/panda#2377] [General]
### Description

When `strictPropertyValues: true` is set in the Panda config, using a CSS variable for a `StrictableProps` is marked as a TS error. This is a relatively reasonable position for Panda to take, but it does mean that the following pattern triggers a TS error, despite it being a valid pattern according to the example at https://panda-css.com/docs/guides/dynamic-styling#using-token:
```
      <div
        className={css({
          display: 'var(--display)',
          boxSizing: 'var(--box-sizing)',
        })}
        style={{
          '--display': 'flex',
          '--box-sizing': 'content-box',
        }}
      >
```

Should `strictPropertyValues` allow the TS template literal type `var(${string})` as a valid value for all `StrictableProps` properties?

Thanks!

### Link to Reproduction

https://play.panda-css.com/gP-PgP1fJE

### Steps to reproduce

1. Open the playground.
2. Observe the typechecking error.

### Panda CSS Version

0.34.3

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

**Comment by astahmer** [comment]
I think not, `strictPropertyValues` job is (at least currently) only about removing the `string & {}` (TS type that allows any string while not removing the literals in a union) that comes from [`csstype`](https://github.com/frenic/csstype) on properties that have a [`syntax`](https://developer.mozilla.org/en-US/docs/Web/CSS/border-top-style#syntax) composed of `keywords`

as alternatives you can:
- use the [escape-hatch syntax](https://panda-css.com/docs/concepts/writing-styles#stricttokens:~:text=a%20config%20token-,For%20one%2Doff%20styles%2C%20you%20can%20always%20use%20the%20escape%2Dhatch%20syntax%20%5Bxxx%5D%20to%20use%20custom%20or%20raw%20CSS%20values%20without%20TypeScript%20errors.,-import%20%7B%20css%20%7D) like so

```ts
      <div
        className={css({
          display: '[var(--display)]',
          boxSizing: '[var(--box-sizing)]',
        })}
        style={{
          '--display': 'flex',
          '--box-sizing': 'content-box',
        }}
      >
```
- this PR will make common CSS variable type-safe https://github.com/chakra-ui/panda/pull/2238

**Reply by etlovett** [reply]
Ok. I think there's a reasonable argument to be made for both approaches here, so fair enough.

One thing that I think would help clarify this for others would be to add an example of the escape-hatch syntax to the code block at https://panda-css.com/docs/concepts/writing-styles#strictpropertyvalues. The `strictTokens` section mentions this syntax, but the `strictPropertyValues` section does not.

**Reply by etlovett** [reply]
Looks like you've decided to support this after all. 😀 https://github.com/chakra-ui/panda/pull/2393 Sounds good!
